src/game/mario_actions_moving.c

```diff
@@ -470,10 +470,11 @@ void update_walking_speed(struct MarioState *m) {
 
 s32 should_begin_sliding(struct MarioState *m) {
     if (m->input & INPUT_ABOVE_SLIDE) {
+        s32 superSlippery = !m->floor ? 0 : m->floor->type == SURFACE_SUPER_SLIPPERY;
```

Comment:
actually last minute nitpick: this can be simplified to just `(m->floor != NULL) && (m->floor->type == SURFACE_SUPER_SLIPPERY)`